src/mixins/filing-template-mixin.ts

```diff
@@ -691,6 +755,70 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
     this.storeStaffPayment(filing)
   }
 
+  /**
+   * Parses a draft special resolution filing into the store.
+   * @param filing the special resolution filing
+   * @param entitySnapshot the latest entity snapshot
+   */
+  parseSpecialResolutionFiling (filing: SpecialResolutionFilingIF, entitySnapshot: EntitySnapshotIF): void {
+    // store Entity Snapshot
+    this.setEntitySnapshot(entitySnapshot)
+
+    // store Entity Type
+    this.setEntityType(filing.alteration.business?.legalType || entitySnapshot.businessInfo.legalType)
+
+    // store Business Information
+    this.setBusinessInformation({
+      ...filing.business,
+      ...filing.alteration.business
+    })
+
+    // store Name Request data
+    this.setNameRequest(
```

Comment:
Not needed?

Reply:
we need this since we are using a name request component, COOP can change a name using name request, correct me I am wrong, I think this is to set namerequest value

Reply:
This is to restore NR data when loading a draft filing.

Again, keep this if you need but I don't know.